### PR TITLE
[W5.7][W16-1]Travis Ching Jia Yea

### PR DESCRIPTION
**Pull request by GilgameshTC**

### CODE REVIEW

**Comment by GilgameshTC** [comment]
Work in progress

**Comment by GilgameshTC** [comment]
Ready for review

**Comment by javenseow** [comment]
Could be improved by not importing commands.* but great job with the PR!

**Comment by kengwoon** [comment]
Not much need for test cases so all is good.

**Comment by pixelky** [comment]
MESSAGE_DUPLICATE_PERSON can be removed from LuckyCommand class. Otherwise, looks good!

**Comment by ericyjw** [comment]
Good job! Could have added comments before the method to explain what the method is doing

File: src/seedu/addressbook/commands/LuckyCommand.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    public static final String COMMAND_WORD = "lucky";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Generates a random 'lucky person' that the "
+            + "user should contact for fun!\n"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "Feeling lucky today? Why not give this person a call!\n"
+            + "Lucky Person: %1$s";
+
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
```
Comment (bqnguyen94) [inline src/seedu/addressbook/commands/LuckyCommand.java]: This is unused?

**Review by bqnguyen94 — CHANGES_REQUESTED** [review]
Looks good, just a minor change needed.

File: src/seedu/addressbook/commands/LuckyCommand.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Generates a random 'lucky person' that the "
+            + "user should contact for fun!\n"
+            + "Parameters: -"
+            + "Example: " + COMMAND_WORD;
+
+    public static final String MESSAGE_SUCCESS = "Feeling lucky today? Why not give this person a call!\n"
+            + "Lucky Person: %1$s";
+
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
```
Comment (pixelky) [inline src/seedu/addressbook/commands/LuckyCommand.java]: This line can be removed. It is not used.

File: test/java/seedu/addressbook/parser/ParserTest.java [hunk 3]
```diff
@@ -81,6 +71,11 @@ public void parse_listCommand_parsedCorrectly() {
         parseAndAssertCommandType(input, ListCommand.class);
     }
 
+    @Test
+    public void parse_luckyCommand_parsedCorrectly() {
```
Comment (ericyjw) [inline test/java/seedu/addressbook/parser/ParserTest.java:75]: Could include a Junit test to test if the Lucky Command generate a random contact, and if all the contact have a probability of getting picked.

Comment (GilgameshTC) [reply test/java/seedu/addressbook/parser/ParserTest.java:75]: Good idea! Will try to implement if I have time.

**Pull request closed without merging**